src/runtime/pkg/device/drivers/block.go

```diff
@@ -60,6 +60,16 @@ func (device *BlockDevice) Attach(ctx context.Context, devReceiver api.DeviceRec
 		return err
 	}
 
+	// With ACRN or firecracker the rootfs for the VM itself sits at /dev/vda and consumes the first
+	// index. Longer term block based VM rootfs should be added as a regular block device which
+	// eliminates the offset.
+	// https://github.com/kata-containers/runtime/issues/1061
+	hypervisorType := devReceiver.GetHypervisorType()
+	if hypervisorType == "acrn" || hypervisorType == "firecracker" {
```

Comment:
@vijaydhanraj Is there any way this condition check could be moved to sandbox.go?

This "device" piece of code isnt the best and will probably need to be refactored at some point, so I would rather have these checks in sandbox part of the code rather than using the "GetHypervisorType" method.

Reply:
@amshinde how about we introduce another interface like `GetHypervisorBlockIndex` that will be implemented in `sandbox.go` which does this check based on `hyperviosrType` and returns an updated index in case of `ACRN` and` Firecracker`?

I am open to other suggestions as well.